pkg/cmd/roachtest/roachtestutil/mixedversion/planner.go

```diff
@@ -446,6 +520,12 @@ func (p *testPlanner) setupTest() (testSetup, []*upgradePlan) {
 
 	var tenantSetup *serviceSetup
 	if p.isMultitenant() {
+		for _, n := range p.currentContext.Tenant.Descriptor.Nodes {
+			handleInternalError(
```

Comment:
Remind me why `changeVersion` here wasn't necessary before this PR? Previously, `changeVersion` was confined to `changeVersionSteps`.

Reply:
This was already "broken" before, but it was inconsequential before the introduction of separate-process tenants.

What this is doing is updating the context associated with the tenant to set the actual released version it is bootstrapped in. The context is initialized in [here](https://github.com/cockroachdb/cockroach/blob/83a68a474a31b6ffc2d7c45d0097b859568e60dc/pkg/cmd/roachtest/roachtestutil/mixedversion/context.go#L168-L199), and the initial version is set to the first binary used in the upgrade test (the one used to [bootstrap the storage cluster](https://github.com/cockroachdb/cockroach/blob/83a68a474a31b6ffc2d7c45d0097b859568e60dc/pkg/cmd/roachtest/roachtestutil/mixedversion/mixedversion.go#L757)). However, tenants are (or can be) started later, potentially after the storage cluster has already gone through a few upgrades.

In separate-process deployments, we display the "released version" for each node in the cluster. Without this change, the version displayed would have been potentially incorrect if bootstrap version != initial version for storage cluster.

Hopefully that helps understand. I added a comment to this code with similar information.